torch_xla_py/xla_model.py

```diff
@@ -497,15 +500,22 @@ def train(self,
         if metrics_debug:
           log_fn(torch_xla._XLAC._xla_metrics_report())
         loss = self._compute_loss(xla_outputs)
+        total_time = time.time() - start_time
```

Comment:
I don't like cramming more stuff into this loop, which is supposed to be easy to read.
Instead, let's create a data structure which holds all the required information, and we pass it to log_fn (if not None).
Then we also supply an API to return a properly formatted string (in the format below), out of the new data structure.
Data structure will be different from train to test.

Then we can have a utility API (in test_util.py) which creates a log_fn, based on other stuff as well, like a writer.

Reply:
I've done something very similar to what you suggested. Let me know what you think!